restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmConfigurationClient.java

```diff
@@ -0,0 +1,528 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2017, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ * For questions related to commercial use licensing, please contact sales@telestax.com.
+ *
+ */
+
+package org.restcomm.android.sdk.fcm;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.restcomm.android.sdk.fcm.model.FcmApplication;
+import org.restcomm.android.sdk.fcm.model.FcmBinding;
+import org.restcomm.android.sdk.fcm.model.FcmCredentials;
+import org.restcomm.android.sdk.util.RCLogger;
+
+import android.util.Base64;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLEncoder;
+
```

Comment:
Let's add a comment with rough description of FcmConfigurationClient

Reply:
Done.